iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -344,4 +364,95 @@ public static PartitionData toPartitionData(StructLike key, Types.StructType key
     }
     return data;
   }
+
+  public static PartitionData toPartitionData(StructLike sourceKey, Types.StructType sourceKeyType,
+      Types.StructType targetKeyType) {
+    PartitionData data = new PartitionData(targetKeyType);
+    for (int i = 0; i < targetKeyType.fields().size(); i++) {
+
+      int fi = i;
+      Object sourceKeyElem = sourceKeyType.fields().stream()
+          .filter(f -> f.name().equals(targetKeyType.fields().get(fi).name())).findFirst().get();
+
+      Object val =  sourceKey.get(sourceKeyType.fields().indexOf(sourceKeyElem),
+          targetKeyType.fields().get(i).type().typeId().javaClass());
+      if (val != null) {
+        data.set(i, val);
+      }
+    }
+    return data;
+  }
+
+  public static Pair<List<DataFile>, List<DeleteFile>> getDataAndDeleteFiles(Table table, int specId,
```

Comment:
why not to split it on 2, instead of this compined Pair<List<DataFile>, List<DeleteFile>>?
are we reusing something and saving cpu cycles?

Reply:
Done, split into 2 methods.